src/metatensor/models/experimental/soap_bpnn/model.py

```diff
@@ -296,27 +290,48 @@ def forward(
 
         soap_features = self.layernorm(soap_features)
 
-        hidden_features = self.bpnn(soap_features)
+        last_layer_features = self.bpnn(soap_features)
+
+        # output the hidden features, if requested:
+        if "mts-models::aux::last_layer_features" in outputs.keys():
+            last_layer_features_options = outputs[
+                "mts-models::aux::last_layer_features"
+            ]
+            out_features = last_layer_features.keys_to_properties(
+                self.center_type_labels.to(device)
+            )
+            if last_layer_features_options.per_atom:
+                # this operation should just remove the center_type label
+                return_dict["mts-models::aux::last_layer_features"] = out_features
+            else:
+                return_dict["mts-models::aux::last_layer_features"] = (
+                    metatensor.torch.sum_over_samples(out_features, ["atom"])
+                )
 
         atomic_energies: Dict[str, TensorMap] = {}
         for output_name, output_layer in self.last_layers.items():
             if output_name in outputs:
                 atomic_energies[output_name] = apply_composition_contribution(
-                    output_layer(hidden_features),
+                    output_layer(last_layer_features),
                     self.composition_weights[  # type: ignore
                         self.output_to_index[output_name]
                     ],
                 )
 
         # Sum the atomic energies coming from the BPNN to get the total energy
-        total_energies: Dict[str, TensorMap] = {}
         for output_name, atomic_energy in atomic_energies.items():
             atomic_energy = atomic_energy.keys_to_samples("center_type")
-            total_energies[output_name] = metatensor.torch.sum_over_samples(
-                atomic_energy, ["atom", "center_type"]
-            )
+            if outputs[output_name].per_atom:
+                # this operation should just remove the center_type label
+                return_dict[output_name] = metatensor.torch.sum_over_samples(
+                    atomic_energy, ["center_type"]
+                )
```

Comment:
this works, but it should be using [`remove_dimension`](https://lab-cosmo.github.io/metatensor/latest/operations/reference/manipulation/manipulate-dimension.html#metatensor.remove_dimension) IMO

Reply:
Thanks, I didn't know